### ISSUE TITLE
Sequelize Warnings with ExpenseItem and PayoutMethod

### ISSUE THREAD

**Issue by znarf** [opencollective/opencollective#5099]
```
(sequelize) Warning: Model "ExpenseItem" is declaring public class fields for attribute(s): "id", "amount", "url", "description", "createdAt", "updatedAt", "deletedAt", "incurredAt", "ExpenseId", "CreatedByUserId".
These class fields are shadowing Sequelize's attribute getters & setters.
See https://sequelize.org/main/manual/model-basics.html#caveat-with-public-class-fields
```

```
(sequelize) Warning: Model "PayoutMethod" is declaring public class fields for attribute(s): "id", "type", "data", "createdAt", "updatedAt", "deletedAt", "name", "isSaved", "CollectiveId", "CreatedByUserId".
These class fields are shadowing Sequelize's attribute getters & setters.
See https://sequelize.org/main/manual/model-basics.html#caveat-with-public-class-fields
```

https://sequelize.org/main/manual/model-basics.html#caveat-with-public-class-fields

Seems this is affecting the classes built with typescript.

**Comment by Betree** [comment]
I think there's no risk on our side since the `restoreSequelizeAttributesOnClass` helper is supposed to restore Sequelize's getters and setters. But there might be a better way to do that nowadays, it's a good time to dig again.

**Comment by Betree** [comment]
We can apparently resolve this with the `declare` typescript keyword, which lets you declare fields on models without overwriting them when they exist. See https://sequelize.org/docs/v6/core-concepts/model-basics/#caveat-with-public-class-fields

**Comment by ephys** [comment]
Note that we've since learned that babel ignores `declare` if the class field is decorated (https://github.com/sequelize/sequelize/issues/14300). `restoreSequelizeAttributesOnClass` is the right solution if `declare` does not work and we can add a flag to mute the warning if it's annoying

**Comment by Betree** [comment]
@ephys Interesting! We don't use decorators right now, so this should work for us. I started a PR to experiment with this in https://github.com/opencollective/opencollective-api/pull/7558; overall things are looking good! I still have an issue in the [`ExpenseItem` model](https://github.com/opencollective/opencollective-api/pull/7558/files#diff-c5faa851968bbd97f06567d996d13d15d7ae203c2cbc2110ac9264cf9a485ec7), where somehow the model attributes still appear as `undefined` even though we use `declare`. I'll dig some more on that.

Great job on the docs BTW, when we first introduced Typescript I remember that Sequelize's docs were not super complete and I was super happy this time with all the explanations and workarounds proposed in https://sequelize.org/docs/v6/other-topics/typescript/.
